_rest_routes.inc.php

```diff
@@ -172,6 +175,14 @@
         RestConfig::authorization_check("patients", "med");
         return (new ListRestController())->delete($pid, $mid, "medical_problem");
     },
+    "GET /api/allergy" => function () {
+        RestConfig::authorization_check("patients", "med");
+        return (new AllergyIntoleranceRestController())->getAll();
+    },
+    "GET /api/allergy/:aid" => function ($uuid) {
+        RestConfig::authorization_check("patients", "med");
+        return (new AllergyIntoleranceRestController())->getOne($uuid);
```

Comment:
to my point above, should be updating/leveraging the calls to ListRestController() rather than creating a new workaround controller.

Reply:
`ListController` is getting used for all the Issues like allergy,medical_problem but for FHIR needs we need to customize all the queries in services for every resource. So using one controller for all might create a mess :thought_balloon:

Reply:
ok to migrate out of the ListController, but we just shouldn't have both mechanisms (ie. shouldn't be able to use new controller and ListController for allergy calls)

Reply:
Agreed for now we can't remove ListController as lots of other endpoints are using that Controller so we can just use New controller for allergy and keep Listcontroller as is :)

Reply:
Will go ahead and update PUT/POST And delete for new Controller that will resolve the issue :)

Reply:
agreed